### PR TITLE
suite-web tests - model R support

### PR DESCRIPTION
**Pull request by mroz22**
- move suite-web end to end tests from `packages/integration-tests/projects/suite-web` to `packages/suite-web/e2e`. 
- update cypress to next major version 10
- refactor js files to ts. this is not 100% completed yet but imho can be done anytime in the future. 
- adds a way how to run emulator from url 
- adds a simple PoC of model R test in suite 

Followups:
- https://github.com/trezor/trezor-suite/issues/6567

### CODE REVIEW

**Comment by tomasklim** [comment]
`browser-not-supported.snap.png`
`chrome-supported-outdated.snap.png`
`firefox-supported-outdated.snap.png`

Should not exist anymore

File: packages/suite-web/e2e/tests/wallet/add-account-types.test.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-// @group:suite
+// @group:wallet
 // @retry=2
 
 import { onAccountsPage } from '../../support/pageObjects/accountsObject';
 import { onSettingsCryptoPage } from '../../support/pageObjects/settingsCryptoObject';
 import { onTopBar } from '../../support/pageObjects/topBarObject';
-import { NetworkSymbol } from '../../../../../../suite-common/wallet-config/src/networksConfig';
+import { NetworkSymbol } from '../../../../../suite-common/wallet-config/src/networksConfig';
```
Comment (tomasklim) [inline packages/suite-web/e2e/tests/wallet/add-account-types.test.ts]:
```suggestion
import { NetworkSymbol } from '@suite-common/wallet-config';
```

File: packages/suite-web/e2e/tests/suite/analytics.test.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // @group:suite
 // @retry=2
 
-import { urlSearchParams } from '../../../../../suite/src/utils/suite/metadata';
+import { urlSearchParams } from '../../../../suite/src/utils/suite/metadata';
```
Comment (tomasklim) [inline packages/suite-web/e2e/tests/suite/analytics.test.ts]: `@suite` or `@trezor/suite`?

File: packages/suite-web/e2e/support/pageObjects/settingsCryptoObject.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /// <reference types="cypress" />
 
-import { NetworkSymbol } from '../../../../../../suite-common/wallet-config/src/networksConfig';
+import { NetworkSymbol } from '../../../../../suite-common/wallet-config/src/networksConfig';
```
Comment (tomasklim) [inline packages/suite-web/e2e/support/pageObjects/settingsCryptoObject.ts]:
```suggestion
import { NetworkSymbol } from '@suite-common/wallet-config';
```

File: packages/suite-web/e2e/support/pageObjects/accountsObject.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /// <reference types="cypress" />
 
-import { NetworkSymbol } from '../../../../../../suite-common/wallet-config/src/networksConfig';
+import { NetworkSymbol } from '../../../../../suite-common/wallet-config/src/networksConfig';
```
Comment (tomasklim) [inline packages/suite-web/e2e/support/pageObjects/accountsObject.ts]:
```suggestion
import { NetworkSymbol } from '@suite-common/wallet-config';
```

File: packages/integration-tests/tsconfig.json [hunk 5]
```diff
@@ -9,13 +9,12 @@
         ],
         "noEmit": true
     },
-    "include": ["**/*.ts", "**/*.js"],
+    "include": [
+        "**/*.ts",
+        "**/*.js",
+        "../suite-web/e2e/process-har.ts"
```
Comment (tomasklim) [inline packages/integration-tests/tsconfig.json]: what is this `integration-tests` package good for now?

File: packages/suite-web/e2e/plugins/bridge.ts [hunk 6]
```diff
@@ -1,5 +1,7 @@
-const express = require('express');
-const EventEmitter = require('events');
+// @ts-nocheck
```
Comment (tomasklim) [inline packages/suite-web/e2e/plugins/bridge.ts:1]: Would be nicer to refactor it instead of // @ts-nocheck

(if it is easy, but should be)

**Review by tomasklim — DISMISSED** [review]

File: packages/integration-tests/tsconfig.json (hunk 5 above)
Comment (mroz22) [reply packages/integration-tests/tsconfig.json]: I dont know yet, at the moment, there are still some tests left (connect-webextension) that are not part of this PR. In the future I am going to either remove it, or turn it into some kind of e2e-common package

**Comment by mroz22** [comment]
> `browser-not-supported.snap.png` `chrome-supported-outdated.snap.png` `firefox-supported-outdated.snap.png`
> 
> Should not exist anymore

You are right. Visual snapshot testing has been broken for a while, in fact inactive. We should not only remove images that are never used but also reactivate it again.

**Comment by mroz22** [comment]
I'd suggest removing all the snapshots completely now and start with a clean slate in another PR. @ondracja ?

File: packages/suite-web/e2e/support/pageObjects/accountsObject.ts (hunk 4 above)
Comment (mroz22) [reply packages/suite-web/e2e/support/pageObjects/accountsObject.ts]: 53156ff7691a6bcc01ae624a200b7b58a56d7cc0

File: packages/suite-web/e2e/support/pageObjects/settingsCryptoObject.ts (hunk 3 above)
Comment (mroz22) [reply packages/suite-web/e2e/support/pageObjects/settingsCryptoObject.ts]: 53156ff7691a6bcc01ae624a200b7b58a56d7cc0

File: packages/suite-web/e2e/tests/suite/analytics.test.ts (hunk 2 above)
Comment (mroz22) [reply packages/suite-web/e2e/tests/suite/analytics.test.ts]: 53156ff7691a6bcc01ae624a200b7b58a56d7cc0

File: packages/suite-web/e2e/tests/wallet/add-account-types.test.ts (hunk 1 above)
Comment (mroz22) [reply packages/suite-web/e2e/tests/wallet/add-account-types.test.ts]: 53156ff7691a6bcc01ae624a200b7b58a56d7cc0

File: packages/suite-web/e2e/plugins/bridge.ts (hunk 6 above)
Comment (mroz22) [reply packages/suite-web/e2e/plugins/bridge.ts:1]: removed no-checks and added some primitive types annotations. Typescript is more like a by-product of this PR, not the main goal, so I  guess it should be enough like this

File: packages/suite-web/e2e/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+    // "extends": "../tsconfig.json",
+    "compilerOptions": {
+        "types": ["cypress", "node"],
+        "esModuleInterop": true,
+        "allowSyntheticDefaultImports": true,
+        "target": "ES3"
```
Comment (tomasklim) [inline packages/suite-web/e2e/tsconfig.json:7]: why ES3?

**Review by tomasklim — COMMENTED** [review]
ignore this comment

**Review by ondracja — APPROVED** [review]

**Comment by ondracja** [comment]
> I'd suggest removing all the snapshots completely now and start with a clean slate in another PR. @ondracja ?

Yes, let's start with a clean slate in another PR.

**Pull request merged**